src/quicklist.c

```diff
@@ -236,55 +241,59 @@ REDIS_STATIC int __quicklistCompressNode(quicklistNode *node) {
         zfree(lzf);
         return 0;
     }
-    lzf = zrealloc(lzf, sizeof(*lzf) + lzf->sz);
-    zfree(node->entry);
+    size_t old_entry_size, new_entry_size;
+    lzf = zrealloc_usable(lzf, sizeof(*lzf) + lzf->sz, &new_entry_size);
+    zfree_usable(node->entry, &old_entry_size);
     node->entry = (unsigned char *)lzf;
     node->encoding = QUICKLIST_NODE_ENCODING_LZF;
+    quicklist->alloc_size += new_entry_size - old_entry_size;
```

Comment:
Because new_entry_size < old_entry_size, we will first obtain a very large value of size_t after substraction, is this an UB?

Reply:
`quicklist->alloc_size` is greater of equal than `old_entry_size` since `old_entry_size` has been accumulated already in `quicklist->alloc_size`.

Wdyt splitting this as below?

```
quicklist->alloc_size -= old_entry_size;
quicklist->alloc_size += new_entry_size;
```

Reply:
Addressed in [db45693](https://github.com/redis/redis/pull/14119/commits/db45693aff8cd33ff3853352585f6754f2e2a4e5).